### PR TITLE
Fix delayed start push

### PR DESCRIPTION
**Pull request by guperrot**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Noticed this while testing something on React Native. Firebase now initialize on its own and can trigger the token update before AppCenter is initialized (which is especially visible with delayed push scenario). And we have an optimization not to send the same token twice but it was used before checking for started/enabled...

### CODE REVIEW

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java [hunk 1]
```diff
@@ -271,16 +271,27 @@ private void enqueuePushInstallationLog(@NonNull String pushToken) {
      */
     @SuppressWarnings("WeakerAccess") /* protected so that Xamarin can use it. */
     protected synchronized void onTokenRefresh(final String pushToken) {
```
Comment (MatkovIvan) [inline sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: @guperrot Should it still be `synchronized`?

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java [hunk 2]
```diff
@@ -271,16 +271,27 @@ private void enqueuePushInstallationLog(@NonNull String pushToken) {
...
+                }
+            };
             post(new Runnable() {
 
                 @Override
                 public void run() {
-                    enqueuePushInstallationLog(pushToken);
+                    if (pushToken != null && !pushToken.equals(mLatestPushToken)) {
+                        AppCenterLog.debug(LOG_TAG, "Push token refreshed: " + pushToken);
+                        mLatestPushToken = pushToken;
```
Comment (MatkovIvan) [inline sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: It isn't in `synchronized` anymore, is it OK?

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: yes for mContext

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java (hunk 2 above)
Comment (guperrot) [reply sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: we access this variable always from the worker thread

**Pull request merged**